src/components/data-table/column-sorting/column-sort.directive.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Directive,
+  Input,
+  HostBinding,
+  HostListener,
+  OnDestroy
+} from "@angular/core";
+import {Subscription} from 'rxjs/Subscription';
+import {
+  SortDirection,
+  IColumnSortingModel
+} from './column-sort';
+import {MdDataColumnSortingService} from './column-sort.service';
+
+/**
+ * @name mdDataSortColumnDirective
+ *
+ * @description
+ * A directive that styles its host with sorting class names, and calls
+ * {@link MdDataColumnSortingService.changeSorting} to the service instance in its
+ * execution context.
+ *
+ * @usage
+ *
+ * <hljs lang="html">
+ *  <tr>
+ *    <th md-data-column-sort="1">Cheese</th>
+ *    <th md-data-column-sort="2">Crackers</th>
+ *    <th md-data-column-sort="3">Wine</th>
+ *  <tr>
+ * </hljs>
+ */
+@Directive({
+  selector: '[md-data-column-sort]'
+})
+export class MdDataColumnSortDirective implements OnDestroy {
+
+  /**
+   * Sorting value of the group of columns.
+   * The group is defined by the injection scope of the {@link MdDataColumnSortingService} in the parent component.
+   */
+  sortingColumn: IColumnSortingModel;
+  sortingSubscription: Subscription;
+
+  /** Unique column identifier */
+  @Input('md-data-column-sort') mdDataSortColumn: string;
+
+  /** Applies 'sortable' CSS class to host */
+  @HostBinding('class.sortable') get sortable() { return true; }
+
+  /** Conditionally applies 'sorted-ascending' CSS class to host */
+  @HostBinding('class.sorted-ascending') get sortingAscending() {
+    return this.isCurrentColumn() && this.sortingColumn.direction === SortDirection.ASCEND;
+  }
+
+  /** Conditionally applies 'sorted-descending' CSS class to host */
+  @HostBinding('class.sorted-descending') get sortingDescending() {
+    return this.isCurrentColumn() && this.sortingColumn.direction === SortDirection.DESCEND;
+  }
+
+  /**
+   * Click handler on host element to tell service to sort by this column
+   * @param clickEvent - the clickEvent.
+   */
+  @HostListener('click', ['$event']) sortBy(clickEvent: Event) {
+    this.sortingService.changeSorting(this.mdDataSortColumn, this.sortingColumn);
+  }
+
+  constructor(private sortingService: MdDataColumnSortingService) {
+    this.sortingSubscription = this.sortingService.sortingColumn$.subscribe(
+        (sorting: IColumnSortingModel) => this.sortingColumn = sorting);
+  }
+
+  ngOnDestroy() {
+    this.sortingSubscription.unsubscribe();
+  }
+
+  private isCurrentColumn():boolean {
+    return this.sortingColumn && this.sortingColumn.column == this.mdDataSortColumn
```

Comment:
used to jslint / tslint catching those for me, but we don't have that going in develop. :stuck_out_tongue: